modules/vstudio/vs2010_vcxproj.lua

```diff
@@ -1477,9 +1477,15 @@
 			elseif (cfg.cppdialect == "C++17") then
 				m.element("LanguageStandard", nil, 'stdcpp17')
 			elseif (cfg.cppdialect == "C++20") then
-				m.element("LanguageStandard", nil, 'stdcpplatest')
+				m.element("LanguageStandard", nil, 'stdcpp20')
 			elseif (cfg.cppdialect == "C++latest") then
 				m.element("LanguageStandard", nil, 'stdcpplatest')
+			elseif _ACTION >= "vs2019" then
+				if (cfg.cdialect == "C11") then
+					m.element("LanguageStandard_C", nil, 'stdc11')
+				elseif (cfg.cdialect == "C17") then
+					m.element("LanguageStandard_C", nil, 'stdc17')
+				end
```

Comment:
These should be in their own function independent of `cppdialect` - this change currently means specifying both `cppdialect` and `cdialect` will result in only the C++ standard to be emit by Premake.